File: src/test/java/cz/vhromada/catalog/web/mapper/TimeMapperTest.java
package cz.vhromada.catalog.web.mapper;

import static org.assertj.core.api.Assertions.assertThat;

import cz.vhromada.catalog.web.common.TimeUtils;
import cz.vhromada.catalog.web.fo.TimeFO;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mapstruct.factory.Mappers;

/**
 * A class represents test for mapper between {@link Integer} and {@link TimeFO}.
 *
 * @author Vladimir Hromada
 */
class TimeMapperTest {

    /**
     * Mapper for time
     */
    private TimeMapper mapper;

    /**
     * Initializes mapper.
     */
    @BeforeEach
    void setUp() {
        mapper = Mappers.getMapper(TimeMapper.class);
    }

    /**
     * Test method for {@link TimeMapper#map(Integer)}.
     */
    @Test
    void map() {
        final Integer length = 100;

        final TimeFO time = mapper.map(length);

        TimeUtils.assertTimeDeepEquals(time, length);
    }

    /**
     * Test method for {@link TimeMapper#map(Integer)} with null length.
     */
    @Test
    void map_NullLength() {
        assertThat(mapper.map(null)).isNull();
    }

    /**
     * Test method for {@link TimeMapper#mapBack(TimeFO)}.
     */
    @Test
    void mapBack() {
        final TimeFO time = TimeUtils.getTimeFO();

        final Integer length = mapper.mapBack(time);

        TimeUtils.assertTimeDeepEquals(time, length);
    }

    /**
     * Test method for {@link TimeMapper#mapBack(TimeFO)} with null time.
     */
    @Test
    void mapBack_NullTime() {
        assertThat(mapper.mapBack(null)).isNull();
    }

}

File: src/test/kotlin/cz/vhromada/catalog/web/mapper/TimeMapperTest.kt
package cz.vhromada.catalog.web.mapper

import cz.vhromada.catalog.web.CatalogMapperTestConfiguration
import cz.vhromada.catalog.web.common.TimeUtils
import cz.vhromada.catalog.web.fo.TimeFO
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * A class represents test for mapper between [Integer] and [TimeFO].
 *
 * @author Vladimir Hromada
 */
@ExtendWith(SpringExtension::class)
@ContextConfiguration(classes = [CatalogMapperTestConfiguration::class])
class TimeMapperTest {

    /**
     * Mapper for time
     */
    @Autowired
    private lateinit var mapper: TimeMapper

    /**
     * Test method for [TimeMapper.map].
     */
    @Test
    fun map() {
        val length = 100

        val time = mapper.map(length)

        TimeUtils.assertTimeDeepEquals(time, length)
    }

    /**
     * Test method for [TimeMapper.mapBack].
     */
    @Test
    fun mapBack() {
        val time = TimeUtils.getTimeFO()

        val length = mapper.mapBack(time)

        TimeUtils.assertTimeDeepEquals(time, length)
    }

}
